packages/core/src/integrations/extraerrordata.ts

```diff
@@ -28,20 +28,22 @@ export class ExtraErrorData implements Integration {
   public setupOnce(): void {
     addGlobalEventProcessor(async (event: SentryEvent, hint?: SentryEventHint) => {
       const self = getCurrentHub().getIntegration(ExtraErrorData);
-
-      if (!self || !hint || !hint.originalException) {
+      if (!self) {
         return event;
       }
-
-      return this.enhanceEventWithErrorData(event, hint.originalException);
+      return self.enhanceEventWithErrorData(event, hint);
     });
   }
 
   /**
    * Attaches extracted information from the Error object to extra field in the SentryEvent
    */
-  public enhanceEventWithErrorData(event: SentryEvent, error: Error): SentryEvent {
-    const errorData = this.extractErrorData(error);
+  public enhanceEventWithErrorData(event: SentryEvent, hint?: SentryEventHint): SentryEvent {
+    if (!hint || !hint.originalException || !(hint.originalException instanceof Error)) {
```

Comment:
Can we use `isError` from `@sentry/utils/is` here instead? It checks for some additional types that are compatible with Error.

Reply:
Sure, done!